src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -2073,22 +2354,74 @@ public BeforePublishEvent(System.IServiceProvider services, ApplicationModel.Dis
     }
 
+    [System.Diagnostics.CodeAnalysis.Experimental("ASPIREPUBLISHERS001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+    public enum CompletionState
+    {
+        InProgress = 0,
+        Completed = 1,
+        CompletedWithWarning = 2,
+        CompletedWithError = 3
+    }
+
+    [System.Diagnostics.CodeAnalysis.Experimental("ASPIREPUBLISHERS001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+    public partial class ContainerBuildOptions
+    {
+        public ContainerImageFormat? ImageFormat { get { throw null; } init { } }
+
+        public string? OutputPath { get { throw null; } init { } }
+
+        public ContainerTargetPlatform? TargetPlatform { get { throw null; } init { } }
+    }
+
+    [System.Diagnostics.CodeAnalysis.Experimental("ASPIREPUBLISHERS001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+    public enum ContainerImageFormat
+    {
+        Docker = 0,
+        Oci = 1
+    }
+
+    [System.Diagnostics.CodeAnalysis.Experimental("ASPIREPUBLISHERS001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+    public enum ContainerTargetPlatform
+    {
+        LinuxAmd64 = 0,
+        LinuxArm64 = 1,
+        LinuxArm = 2,
+        Linux386 = 3,
+        WindowsAmd64 = 4,
+        WindowsArm64 = 5
+    }
+
     public partial interface IDistributedApplicationPublisher
     {
         System.Threading.Tasks.Task PublishAsync(ApplicationModel.DistributedApplicationModel model, System.Threading.CancellationToken cancellationToken);
     }
 
     [System.Diagnostics.CodeAnalysis.Experimental("ASPIREPUBLISHERS001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
-    public partial interface IPublishingActivityProgressReporter
+    public partial interface IPublishingActivityReporter
+    {
+        System.Threading.Tasks.Task CompletePublishAsync(string? completionMessage = null, CompletionState? completionState = null, bool isDeploy = false, System.Threading.CancellationToken cancellationToken = default);
+        System.Threading.Tasks.Task<IPublishingStep> CreateStepAsync(string title, System.Threading.CancellationToken cancellationToken = default);
+    }
+
+    [System.Diagnostics.CodeAnalysis.Experimental("ASPIREPUBLISHERS001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+    public partial interface IPublishingStep : System.IAsyncDisposable
+    {
+        System.Threading.Tasks.Task CompleteAsync(string completionText, CompletionState completionState = CompletionState.Completed, System.Threading.CancellationToken cancellationToken = default);
+        System.Threading.Tasks.Task<IPublishingTask> CreateTaskAsync(string statusText, System.Threading.CancellationToken cancellationToken = default);
+    }
+
+    [System.Diagnostics.CodeAnalysis.Experimental("ASPIREPUBLISHERS001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+    public partial interface IPublishingTask : System.IAsyncDisposable
```

Comment:
Consider renaming this to `IPipelineTask`, as it's used in both publish and deploy.